src/clientManager.js

```diff
@@ -94,28 +94,34 @@ ClientManager.prototype.addUser = function(name, password) {
 	return true;
 };
 
-ClientManager.prototype.updateUser = function(name, opts) {
-	var users = this.getUsers();
+ClientManager.prototype.updateUser = function(name, opts, callback) {
+	const users = this.getUsers();
 	if (users.indexOf(name) === -1) {
 		return false;
 	}
 	if (typeof opts === "undefined") {
 		return false;
 	}
 
-	var user = {};
-	try {
-		user = this.readUserConfig(name);
-		_.assign(user, opts);
-		fs.writeFileSync(
-			Helper.getUserConfigPath(name),
-			JSON.stringify(user, null, "\t")
-		);
-	} catch (e) {
-		log.error("Failed to update user", e);
-		return;
+	let user = this.readUserConfig(name);
+	const currentUser = JSON.stringify(user, null, "\t");
```

Comment:
Yeah, this isn't ideal, and could cause some false positives (if the user's properties are now in a different order, but exactly the same) but I doubt that's going to cause any problems.

And it's definitely the fastest way of comparing them, it's just not the most accurate.